src/main/java/org/java_websocket/client/WebSocketClient.java

```diff
@@ -480,10 +484,17 @@ public void run() {
       }
 
       // if the socket is set by others we don't apply any TLS wrapper
-      if (isNewSocket && "wss".equals(uri.getScheme())) {
-        SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
-        sslContext.init(null, null, null);
-        SSLSocketFactory factory = sslContext.getSocketFactory();
+      if (upgradeSocketToSSLSocket && "wss".equals(uri.getScheme())) {
+        SSLSocketFactory factory = null;
```

Comment:
I would remove the initialization with null to make the compiler check that factory is initialized in all code branches.